### PR TITLE
Hide problem weight when set to zero

### PR DESCRIPTION
**Pull request by azimpradhan**
```
This PR hides the problem weight when the problem
weight is set to zero. This makes the free-text-xblock
follow the convention used by problem types, which also
hide the "0 points possible" text instead of displaying
it. A test of this feature has been added to the suite
of tests for this xblock.
```

### CODE REVIEW

**Comment by azimpradhan** [comment]
@mondiaz, @stvstnfrd, @caesar2164, this PR is ready for review

**Comment by azimpradhan** [comment]
@stvstnfrd, @caesar2164, this is ready for review. For this one only look at the topmost commit: 83693a2

File: freetextresponse/private/view.html [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <div class="freetextresponse">
     <h2 class="problem-header">{self.display_name}</h2>
-    <div class="problem-progress">({problem_progress})</div>
+    <div class="problem-progress">{problem_progress}</div>
     <p>{self.prompt}</p>
     <div class="user-input {indicator_class}">
         <textarea class="student_answer" style="height:150px" rows="20" cols="70">{self.student_answer}</textarea>
```
Comment (caesar2164) [inline freetextresponse/private/view.html:6]: don't love the inline style...

File: freetextresponse/tests.py [hunk 2]
```diff
@@ -348,7 +358,10 @@ def test_problem_progress_score_positive_weight_singular(self):
         """
         self.xblock.score = 1
         self.xblock.weight = 1
-        self.assertEquals(_('1/1 point'), self.xblock._get_problem_progress())
+        self.assertEquals(
+            _('(1/1 point)'),
```
Comment (caesar2164) [inline freetextresponse/tests.py:362]: did you mean to keep the () inside the string here?

File: freetextresponse/freetextresponse.py [hunk 3]
```diff
@@ -361,24 +361,29 @@ def _get_problem_progress(self):
...
+        if self.weight == 0:
+            result = ''
+        elif self.score == 0.0:
+            result = "({})".format(
+                ungettext(
+                    "{weight} point possible",
+                    "{weight} points possible",
+                    self.weight,
+                ).format(
+                    weight=self.weight
```
Comment (stvstnfrd) [inline freetextresponse/freetextresponse.py]: comma

**Comment by stvstnfrd** [comment]
lgtm

File: freetextresponse/tests.py (hunk 2 above)
Comment (azimpradhan) [inline freetextresponse/tests.py:362]: yes

**Pull request merged**